src/idl_parser.cpp

```diff
@@ -918,6 +918,10 @@ CheckedError Parser::ParseField(StructDef &struct_def) {
   ECHECK(ParseType(type));
 
   if (struct_def.fixed) {
+    if (IsIncompleteStruct(type)
+        || (IsArray(type) && IsIncompleteStruct(type.VectorType())))
+        return Error("Incomplete type in struct is not allowed");
```

Comment:
Can we print out the type name in the error message as well?

Reply:
Done